test/e2e/ocpbugs48050_test.go

```diff
@@ -0,0 +1,611 @@
+	if err := kclient.Get(context.TODO(), ingressOperator, &deployment); err != nil {
+		return "", fmt.Errorf("failed to get deployment %s/%s: %v", ingressOperator.Namespace, ingressOperator.Name, err)
+	}
+
+	for _, container := range deployment.Spec.Template.Spec.Containers {
+		for _, env := range container.Env {
+			if env.Name == "CANARY_IMAGE" {
+				return env.Value, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("CANARY_IMAGE environment variable not found in deployment %s/%s", ingressOperator.Namespace, ingressOperator.Name)
+}
+
+func queryPrometheusForDuplicateTEHeaders(promClient prometheusv1client.API, namespace string, timeout time.Duration) (map[string]float64, error) {
+	query := fmt.Sprintf(`sum by (route) (haproxy_backend_duplicate_te_header_total{exported_namespace="%s"})`, namespace)
+
+	result, err := queryPrometheus(promClient, query, timeout)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query Prometheus for duplicate TE headers: %w", err)
+	}
+
+	vector, ok := result.(model.Vector)
+	if !ok {
+		return nil, fmt.Errorf("expected result type model.Vector, got %T", result)
+	}
+
+	routeCounts := make(map[string]float64)
+	for _, sample := range vector {
+		routeName := string(sample.Metric["route"])
+		value := float64(sample.Value)
+		routeCounts[routeName] = value
+	}
+
+	return routeCounts, nil
+}
+
+// TestOCPBUGS48050 validates the new metric [1],
+// duplicate_te_header_total, which detects duplicate
+// Transfer-Encoding headers per backend response. This test addresses
+// issues encountered when upgrading from HAProxy 2.2 in OpenShift
+// 4.13 to HAProxy 2.6 in OpenShift 4.14, where responses are rejected
+// due to RFC 7230 compliance.
+//
+// [1] https://github.com/openshift/router/pull/626.
+func TestOCPBUGS48050(t *testing.T) {
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ns, err := setupOCPBUGS48050(t, 10)
+	if err != nil {
+		t.Fatalf("failed to setup test resources: %v", err)
+	}
+
+	routeList, err := waitForAllRoutesAdmitted(ns.Name, 2*time.Minute, func(admittedRoutes, totalRoutes int, pendingRoutes []string) {
+		if len(pendingRoutes) > 0 {
+			t.Logf("%d/%d routes admitted. Waiting for: %s", admittedRoutes, totalRoutes, strings.Join(pendingRoutes, ", "))
+		} else {
+			t.Logf("All %d routes in namespace %s have been admitted", totalRoutes, ns.Name)
+		}
+	})
+	if err != nil {
+		t.Fatalf("Error waiting for routes to be admitted: %v", err)
+	}
+
+	httpClient := &http.Client{
+		Timeout: time.Minute,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	httpGet := func(url string, check func(*http.Response, error) error) error {
+		t.Logf("GET %s", url)
+		resp, err := httpClient.Get(url)
+		if resp != nil {
+			defer func(Body io.ReadCloser) {
+				_ = Body.Close()
+			}(resp.Body)
```

Comment:
nit: I think this should work too:
```suggestion
			defer resp.Body.Close()
```

Reply:
It does. I just accepts Golang's inspection which now hints that we don't care for the return values from Close().

Reply:
It's supposed to say "I'm aware that calls to Close() can fail, but in this instance I don't care".

Reply:
> It's supposed to say "I'm aware that calls to Close() can fail, but in this instance I don't care".

Right, I thought that since we don't care we could shorten this one.